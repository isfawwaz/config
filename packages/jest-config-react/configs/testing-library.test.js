const withRTL = require('./testing-library');

describe('Testing Library', () => {
  const buildConfig = () => ({
    testEnvironment: 'jsdom',
    setupFilesAfterEnv: ['foo'],
  });

  it('should match snapshot', () => {
    expect(withRTL()(buildConfig())).toMatchSnapshot();
  });

  it('should throw if testEnvironment is not jsdom', () => {
    const config = {
      ...buildConfig(),
      testEnvironment: 'node',
    };

    expect(() => withRTL()(config)).toThrow(/expected testEnvironment to be set to "jsdom"/i);
  });
});
