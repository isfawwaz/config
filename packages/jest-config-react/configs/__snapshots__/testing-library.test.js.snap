// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Library should match snapshot 1`] = `
{
  "setupFilesAfterEnv": [
    "foo",
    "<HOME_DIR>/Programming/web/self-dev/personal-config/node_modules/.pnpm/@testing-library+jest-dom@5.16.5/node_modules/@testing-library/jest-dom/extend-expect.js",
  ],
  "testEnvironment": "jsdom",
}
`;
